sql: wrap stacktraceless errors with errors.Wrap

ecwall opened the pull request:
Fixes #95794

This replaces the previous attempt to add logging here #95797.

The context itself cannot be augmented to add a stack trace to errors because
it interferes with grpc timeout logic - gRPC compares errors directly without
checking causes https://github.com/grpc/grpc-go/blob/v1.46.0/rpc_util.go#L833.
Although the method signature allows it, `Context.Err()` should not be
overriden to customize the error:
```
// If Done is not yet closed, Err returns nil.
// If Done is closed, Err returns a non-nil error explaining why:
// Canceled if the context was canceled
// or DeadlineExceeded if the context's deadline passed.
// After Err returns a non-nil error, successive calls to Err return the same error.
Err() error
```
Additionally, a child context of the augmented context may end up being used
which will circumvent the stack trace capture.

This change instead wraps `errors.Wrap` in a few places that might end up
helping debug the original problem:
1) Where we call `Context.Err()` directly.
2) Where gRPC returns an error after possibly calling `Context.Err()`
   internally or returns an error that does not have a stack trace.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/96659)
<!-- Reviewable:end -->

andreimatei reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96659)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ecwall and @rafiss)

___
*[`-- commits` line 8 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNbxf_65MgkkIdCAtzR):*
nit: talk primarily about the contract of `Context.Err()`.
___
*[`-- commits` line 11 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNbxjX06D42VehMhF6C):*
Can you try to also put messages on many of the points you touch? (i.e. `errors.Wrap{f}()`).
___
*[`pkg/roachpb/errors.go` line 323 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNbwA9LEVsz-zmPpvg_:-NNbwA9M0v5F8pdF5khf:b-hzrrfe) ([raw file](https://github.com/cockroachdb/cockroach/blob/76025b855cdb9297a766903f9df92d6c1649cf03/pkg/roachpb/errors.go#L323)):*
> ```Go
> 				// TODO(tbg): revisit this unintuitive error wrapping here and see if
> 				// a better solution can be found.
> 				return errors.WithStack(&UnhandledRetryableError{
> ```

I think the changes to this function are dubious. We're not interested in the stack trace of the caller to `GoError`.
I think you want to capture a stack trace when this `Error` is returned from a `Batch` RPC. In fact, in the interesting case of a timeout error returned from client-side gRPC, I think you'll already have a better stack trace in the error given your changes to `TracingInternalClient` (I think `DistSender` takes the error returned by the RPC (and now modified in `TracingInternalClient`) and turns it into an `roachpb.Error` using `roachpb.NewError(err))`) , and in that case adding another one here I think can only hurt.
I'd try to capture the backtrace around `grpcTransport` (and overwrite `EncodedError`). Btw, I think including the request in the error (`ba.String()`) would also be useful.

If you're not attracted to doing that, I'd suggest we don't touch this code at all, as it's not relevant for cancelation or timeout.
___
*[`pkg/rpc/nodedialer/nodedialer.go` line 321 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNc-WBmDjL1tfqDv_2T:-NNc-WBmDjL1tfqDv_2U:b-mmmog6) ([raw file](https://github.com/cockroachdb/cockroach/blob/332fd3045a10cda1b248abf11da3f5a503c7ee87/pkg/rpc/nodedialer/nodedialer.go#L321)):*
> ```Go
> 	resp, err := tic.InternalClient.Batch(ctx, ba, opts...)
> 	if err != nil {
> 		return nil, errors.WithStack(err)
> ```

If we do something for the `BatchResponse` in `grpcTransport` like I suggest elsewhere, I think maybe wrapping this `err` should also move there for symmetry. In fact even without those other changes, I think this `TracingInternalClient` would be better left alone to deal with one thing only, and lift the error mangling up.
___
*[`pkg/util/tracing/grpcinterceptor/grpc_interceptor.go` line 97 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNbzHdv5cMHGWOmCPYw:-NNbzHdv5cMHGWOmCPYx:bvkhu5f) ([raw file](https://github.com/cockroachdb/cockroach/blob/76025b855cdb9297a766903f9df92d6c1649cf03/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go#L97)):*
> ```Go
> 		if methodExcludedFromTracing(info.FullMethod) {
> 			resp, err := handler(ctx, req)
> 			return resp, errors.WithStack(err)
> ```

The server-side stack that you're capturing here will always look the same (at least when the `internalClientAdapter` is not used)... Also I don't think it will make it across gRPC; in fact I suspect you might break the grpc return codes.
I don't think you want to capture it here.
___
*[`pkg/util/tracing/grpcinterceptor/grpc_interceptor.go` line 229 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNcAddC8MyNJqemMKRJ:-NNcAddC8MyNJqemMKRK:bcfyzzs) ([raw file](https://github.com/cockroachdb/cockroach/blob/332fd3045a10cda1b248abf11da3f5a503c7ee87/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go#L229)):*
> ```Go
> 		_, localRequest := grpcutil.IsLocalRequestContext(ctx)
> 		if localRequest {
> 			return errors.WithStack(invoker(ctx, method, req, resp, cc, opts...))
> ```

I wouldn't put more logic into this tracing interceptor. I'd rather we add another interceptor around [here](https://github.com/cockroachdb/cockroach/blob/79e3bd7d2f3be013bc9592334ce24748888b339f/pkg/rpc/context.go#L665).
Is there a high bar to doing that for streaming RPCs in particular? If so, I'd leave those alone, without the error wrapping.


<!-- Sent from Reviewable.io -->

ecwall committed:
sql: wrap stacktraceless errors with errors.Wrap

Fixes #95794

This replaces the previous attempt to add logging here #95797.

The context itself cannot be augmented to add a stack trace to errors because
it interferes with grpc timeout logic - gRPC compares errors directly without
checking causes https://github.com/grpc/grpc-go/blob/v1.46.0/rpc_util.go#L833.
Although the method signature allows it, `Context.Err()` should not be
overriden to customize the error:
```
// If Done is not yet closed, Err returns nil.
// If Done is closed, Err returns a non-nil error explaining why:
// Canceled if the context was canceled
// or DeadlineExceeded if the context's deadline passed.
// After Err returns a non-nil error, successive calls to Err return the same error.
Err() error
```
Additionally, a child context of the augmented context may end up being used
which will circumvent the stack trace capture.

This change instead wraps `errors.Wrap` in a few places that might end up
helping debug the original problem:
1) Where we call `Context.Err()` directly.
2) Where gRPC returns an error after possibly calling `Context.Err()`
   internally or returns an error that does not have a stack trace.

Release note: None

ecwall reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96659)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei and @rafiss)

___
*[`-- commits` line 8 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNbxf_65MgkkIdCAtzR):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

nit: talk primarily about the contract of `Context.Err()`.
</blockquote></details>

Added some more details.
___
*[`-- commits` line 11 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNbxjX06D42VehMhF6C):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

Can you try to also put messages on many of the points you touch? (i.e. `errors.Wrap{f}()`).
</blockquote></details>

Yeah I changed `WithStack` -> `Wrap`.
___
*[`pkg/rpc/nodedialer/nodedialer.go` line 321 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNc-WBmDjL1tfqDv_2T:-NNcpZrC--oILlJv4R_v:b10dzml) ([raw file](https://github.com/cockroachdb/cockroach/blob/332fd3045a10cda1b248abf11da3f5a503c7ee87/pkg/rpc/nodedialer/nodedialer.go#L321)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

If we do something for the `BatchResponse` in `grpcTransport` like I suggest elsewhere, I think maybe wrapping this `err` should also move there for symmetry. In fact even without those other changes, I think this `TracingInternalClient` would be better left alone to deal with one thing only, and lift the error mangling up.
</blockquote></details>

Reverted this back.
___
*[`pkg/util/tracing/grpcinterceptor/grpc_interceptor.go` line 97 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNbzHdv5cMHGWOmCPYw:-NNcpi2QA9t56lCAButy:bg6tzi0) ([raw file](https://github.com/cockroachdb/cockroach/blob/76025b855cdb9297a766903f9df92d6c1649cf03/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go#L97)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

The server-side stack that you're capturing here will always look the same (at least when the `internalClientAdapter` is not used)... Also I don't think it will make it across gRPC; in fact I suspect you might break the grpc return codes.
I don't think you want to capture it here.
</blockquote></details>

Removed changes in the interceptors.
___
*[`pkg/util/tracing/grpcinterceptor/grpc_interceptor.go` line 229 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNcAddC8MyNJqemMKRJ:-NNcpqQkD3o6CysPXKeI:b-2kxsi1) ([raw file](https://github.com/cockroachdb/cockroach/blob/332fd3045a10cda1b248abf11da3f5a503c7ee87/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go#L229)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

I wouldn't put more logic into this tracing interceptor. I'd rather we add another interceptor around [here](https://github.com/cockroachdb/cockroach/blob/79e3bd7d2f3be013bc9592334ce24748888b339f/pkg/rpc/context.go#L665).
Is there a high bar to doing that for streaming RPCs in particular? If so, I'd leave those alone, without the error wrapping.
</blockquote></details>

I moved the changes here to the spot you linked to and conditionally wrap if it is not a status error to avoid interfering with the caller. I am not sure if we need this or not... can stacktraceless errors come from here?
___
*[`pkg/roachpb/errors.go` line 323 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNbwA9LEVsz-zmPpvg_:-NNcXBMO-B-gVk25dQK2:b10dzml) ([raw file](https://github.com/cockroachdb/cockroach/blob/76025b855cdb9297a766903f9df92d6c1649cf03/pkg/roachpb/errors.go#L323)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

I think the changes to this function are dubious. We're not interested in the stack trace of the caller to `GoError`.
I think you want to capture a stack trace when this `Error` is returned from a `Batch` RPC. In fact, in the interesting case of a timeout error returned from client-side gRPC, I think you'll already have a better stack trace in the error given your changes to `TracingInternalClient` (I think `DistSender` takes the error returned by the RPC (and now modified in `TracingInternalClient`) and turns it into an `roachpb.Error` using `roachpb.NewError(err))`) , and in that case adding another one here I think can only hurt.
I'd try to capture the backtrace around `grpcTransport` (and overwrite `EncodedError`). Btw, I think including the request in the error (`ba.String()`) would also be useful.

If you're not attracted to doing that, I'd suggest we don't touch this code at all, as it's not relevant for cancelation or timeout.

</blockquote></details>

Reverted this back.


<!-- Sent from Reviewable.io -->

andreimatei reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96659)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ecwall and @rafiss)

___
*[`-- commits` line 11 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNbxjX06D42VehMhF6C):*
<details><summary><i>Previously, ecwall (Evan Wall) wrote…</i></summary><blockquote>

Yeah I changed `WithStack` -> `Wrap`.
</blockquote></details>

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 
___
*[`pkg/ccl/kvccl/kvtenantccl/connector.go` line 464 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNia1AZ4MGIvcHVIIgU:-NNia1AZ4MGIvcHVIIgV:b-mtdg6l) ([raw file](https://github.com/cockroachdb/cockroach/blob/063dd3b3cff20afdbcebcd78af8bfdce7b907207/pkg/ccl/kvccl/kvtenantccl/connector.go#L464)):*
> ```Go
> 		return resp.Descriptors, resp.PrefetchedDescriptors, nil
> 	}
> 	return nil, nil, errors.Wrap(ctx.Err(), "range lookup context error")
> ```

nit: consider scratching "context error" from everywhere; I think the message of the two context errors is very recognizable
___
*[`pkg/kv/kvclient/kvcoord/transport.go` line 238 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNiaRgd-wOo3Ef2TsqN:-NNiaRgd-wOo3Ef2TsqO:b-w4elw2) ([raw file](https://github.com/cockroachdb/cockroach/blob/063dd3b3cff20afdbcebcd78af8bfdce7b907207/pkg/kv/kvclient/kvcoord/transport.go#L238)):*
> ```Go
> 		span.ImportRemoteRecording(reply.CollectedSpans)
> 	}
> 	return reply, errors.Wrapf(err, "send batch error %s", ba.String())
> ```

Let's do the `ba.String()` only if `err != nil`; it does cost.
___
*[`pkg/kv/kvclient/kvcoord/transport.go` line 238 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNiacPd7tzrQFoXK77E:-NNiacPd7tzrQFoXK77F:bvfvwcu) ([raw file](https://github.com/cockroachdb/cockroach/blob/063dd3b3cff20afdbcebcd78af8bfdce7b907207/pkg/kv/kvclient/kvcoord/transport.go#L238)):*
> ```Go
> 		span.ImportRemoteRecording(reply.CollectedSpans)
> 	}
> 	return reply, errors.Wrapf(err, "send batch error %s", ba.String())
> ```

let's try to move the `ba` before the word "error", otherwise it'll be confusing where the error message starts. 
Maybe `ba: %s RPC error`.
___
*[`pkg/rpc/context.go` line 674 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNibO_F-VmGQUmDuKFq:-NNibO_F-VmGQUmDuKFr:b-n5ox0h) ([raw file](https://github.com/cockroachdb/cockroach/blob/063dd3b3cff20afdbcebcd78af8bfdce7b907207/pkg/rpc/context.go#L674)):*
> ```Go
> 				if !ok {
> 					// Wrapping status errors interferes with caller.
> 					err = errors.Wrap(err, "client interceptor error")
> ```

this message is confusing; the error is not coming from "client interceptor"; it's coming from an RPC.
___
*[`pkg/rpc/context.go` line 682 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNihQqlEDHNf0VeFKqk:-NNihQqlEDHNf0VeFKql:b-d5g0o8) ([raw file](https://github.com/cockroachdb/cockroach/blob/063dd3b3cff20afdbcebcd78af8bfdce7b907207/pkg/rpc/context.go#L682)):*
> ```Go
> 		rpcCtx.clientStreamInterceptors = append(
> 			rpcCtx.clientStreamInterceptors,
> 			func(ctx context.Context, desc *grpc.StreamDesc, cc *grpc.ClientConn, method string, streamer grpc.Streamer, opts ...grpc.CallOption) (grpc.ClientStream, error) {
> ```

If it was me, I'd leave streaming RPCs alone. Usually the timeouts around them are encountered when reading or writing to the stream, not when initiating the call (i.e. here). You don't go through the trouble of providing a custom implementing a stream implementation that deals with that, so I think adding this code here doesn't serve much.
___
*[`pkg/server/init.go` line 480 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNid5C6BOUviywVOhUR:-NNid5C6BOUviywVOhUS:b-lha9wa) ([raw file](https://github.com/cockroachdb/cockroach/blob/063dd3b3cff20afdbcebcd78af8bfdce7b907207/pkg/server/init.go#L480)):*
> ```Go
> 		status, ok := grpcstatus.FromError(errors.UnwrapAll(err))
> 		if !ok {
> 			return nil, errors.Wrap(err, "non-grpc join error")
> ```

what is the "non-grpc" trying to tell me?
Maybe "failed to join cluster".
___
*[`pkg/sql/internal_result_channel.go` line 114 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNie3ZYBPehEp_Ewsuc:-NNie3ZYBPehEp_Ewsud:b-61nvx2) ([raw file](https://github.com/cockroachdb/cockroach/blob/063dd3b3cff20afdbcebcd78af8bfdce7b907207/pkg/sql/internal_result_channel.go#L114)):*
> ```Go
> 	ctx context.Context,
> ) (_ ieIteratorResult, done bool, err error) {
> 	const wrapMsg = "first result context error"
> ```

maybe "failed to read query result"
___
*[`pkg/sql/internal_result_channel.go` line 132 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNieTKH-ltaEVmHzV-J:-NNieTKH-ltaEVmHzV-K:ba99ifx) ([raw file](https://github.com/cockroachdb/cockroach/blob/063dd3b3cff20afdbcebcd78af8bfdce7b907207/pkg/sql/internal_result_channel.go#L132)):*
> ```Go
> 		return false, nil
> 	}
> 	const wrapMsg = "maybe unblock writer context error"
> ```

Up to you, but I feel like the amount of changes has gone overboard for little chance of actually helping anybody.
There's also some readability cost to lines like
```
	case <-i.doneCh:
		return true, errors.Wrap(ctx.Err(), wrapMsg)
```
Before, it was clear enough that the error returned could be `nil`; now that's less visible.

A lot of these call sites deal mostly with cancelations, not with timeouts. And for cancelation, I think the `Cause(ctx)` is what we want (once we also start using `WithCancelCause`).

So it it was me, I wouldn't blindly wrap every `ctx.Err()` that I can find.
___
*[`pkg/util/tracing/grpcinterceptor/grpc_interceptor.go` line 229 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNcAddC8MyNJqemMKRJ:-NNigihz3ScJgXH4IkGI:b-hncs4o) ([raw file](https://github.com/cockroachdb/cockroach/blob/332fd3045a10cda1b248abf11da3f5a503c7ee87/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go#L229)):*
> I am not sure if we need this or not... can stacktraceless errors come from here?

Sorry - what's "here"?
___
*[`pkg/util/tracing/grpcinterceptor/grpc_interceptor.go` line 253 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNigZfo38IlmJmVDttM:-NNigZfo38IlmJmVDttN:b-3n0op8) ([raw file](https://github.com/cockroachdb/cockroach/blob/063dd3b3cff20afdbcebcd78af8bfdce7b907207/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go#L253)):*
> ```Go
> 				clientSpan.Recordf("error: %s", err)
> 			}
> 			return err
> ```

if you're taking the blame here, move this `return` in the `if` above`.


<!-- Sent from Reviewable.io -->

ecwall reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96659)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @andreimatei and @rafiss)

___
*[`pkg/ccl/kvccl/kvtenantccl/connector.go` line 464 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNia1AZ4MGIvcHVIIgU:-NNlIk086JZZY1vTS6UX:ba3dfob) ([raw file](https://github.com/cockroachdb/cockroach/blob/063dd3b3cff20afdbcebcd78af8bfdce7b907207/pkg/ccl/kvccl/kvtenantccl/connector.go#L464)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

nit: consider scratching "context error" from everywhere; I think the message of the two context errors is very recognizable
</blockquote></details>

Removed
___
*[`pkg/kv/kvclient/kvcoord/transport.go` line 238 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNiaRgd-wOo3Ef2TsqN:-NNlJ6JVC0KnogE9uTIA:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/063dd3b3cff20afdbcebcd78af8bfdce7b907207/pkg/kv/kvclient/kvcoord/transport.go#L238)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

Let's do the `ba.String()` only if `err != nil`; it does cost.
</blockquote></details>

Done
___
*[`pkg/kv/kvclient/kvcoord/transport.go` line 238 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNiacPd7tzrQFoXK77E:-NNlJ7i021rqpPrIHzNG:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/063dd3b3cff20afdbcebcd78af8bfdce7b907207/pkg/kv/kvclient/kvcoord/transport.go#L238)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

let's try to move the `ba` before the word "error", otherwise it'll be confusing where the error message starts. 
Maybe `ba: %s RPC error`.
</blockquote></details>

Done
___
*[`pkg/rpc/context.go` line 674 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNibO_F-VmGQUmDuKFq:-NNlJGaV3B2LKscUS4tQ:btgb6mi) ([raw file](https://github.com/cockroachdb/cockroach/blob/063dd3b3cff20afdbcebcd78af8bfdce7b907207/pkg/rpc/context.go#L674)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

this message is confusing; the error is not coming from "client interceptor"; it's coming from an RPC.
</blockquote></details>

Removed this based on the comment below
___
*[`pkg/rpc/context.go` line 682 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNihQqlEDHNf0VeFKqk:-NNlJIztEY9bOy0t_iev:b192ohd) ([raw file](https://github.com/cockroachdb/cockroach/blob/063dd3b3cff20afdbcebcd78af8bfdce7b907207/pkg/rpc/context.go#L682)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

If it was me, I'd leave streaming RPCs alone. Usually the timeouts around them are encountered when reading or writing to the stream, not when initiating the call (i.e. here). You don't go through the trouble of providing a custom implementing a stream implementation that deals with that, so I think adding this code here doesn't serve much.
</blockquote></details>

Removed these impls
___
*[`pkg/server/init.go` line 480 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNid5C6BOUviywVOhUR:-NNlK7l-AeWGBynmt_LC:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/063dd3b3cff20afdbcebcd78af8bfdce7b907207/pkg/server/init.go#L480)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

what is the "non-grpc" trying to tell me?
Maybe "failed to join cluster".
</blockquote></details>

Done
___
*[`pkg/sql/internal_result_channel.go` line 114 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNie3ZYBPehEp_Ewsuc:-NNlKEZ-Bo16zIhkgsGM:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/063dd3b3cff20afdbcebcd78af8bfdce7b907207/pkg/sql/internal_result_channel.go#L114)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

maybe "failed to read query result"
</blockquote></details>

Done
___
*[`pkg/sql/internal_result_channel.go` line 132 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNieTKH-ltaEVmHzV-J:-NNlY3KB8uynXy4v0KF2:bhhlgp3) ([raw file](https://github.com/cockroachdb/cockroach/blob/063dd3b3cff20afdbcebcd78af8bfdce7b907207/pkg/sql/internal_result_channel.go#L132)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

Up to you, but I feel like the amount of changes has gone overboard for little chance of actually helping anybody.
There's also some readability cost to lines like
```
	case <-i.doneCh:
		return true, errors.Wrap(ctx.Err(), wrapMsg)
```
Before, it was clear enough that the error returned could be `nil`; now that's less visible.

A lot of these call sites deal mostly with cancelations, not with timeouts. And for cancelation, I think the `Cause(ctx)` is what we want (once we also start using `WithCancelCause`).

So it it was me, I wouldn't blindly wrap every `ctx.Err()` that I can find.
</blockquote></details>

The reason I added these is that it looks like any of these could be where `ctx.Err()` is first discovered when using the `InternalExecutor`. I added a comment about `errors.Wrap` returning `nil` if `nil` is passed in.
___
*[`pkg/util/tracing/grpcinterceptor/grpc_interceptor.go` line 229 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNcAddC8MyNJqemMKRJ:-NNlLeSO5dS1gZq-BSFf:b-rp2nyd) ([raw file](https://github.com/cockroachdb/cockroach/blob/332fd3045a10cda1b248abf11da3f5a503c7ee87/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go#L229)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

> I am not sure if we need this or not... can stacktraceless errors come from here?

Sorry - what's "here"?
</blockquote></details>

I removed this anyways, but I meant from inside `invoker`.
___
*[`pkg/util/tracing/grpcinterceptor/grpc_interceptor.go` line 253 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-NNigZfo38IlmJmVDttM:-NNlM-Ez7yBKEE-77bOT:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/063dd3b3cff20afdbcebcd78af8bfdce7b907207/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go#L253)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

if you're taking the blame here, move this `return` in the `if` above`.
</blockquote></details>

Done


<!-- Sent from Reviewable.io -->

andreimatei reviewed (approved):
Merge away as far as I'm concerned.
I don't know about backporting, though. I'd rather we just revert the previous change on the other branches. Do we really need to backport anything?

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-:-NNlrgV53p74TATagKjb:b1x5os8)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @ecwall and @rafiss)



<!-- Sent from Reviewable.io -->

ecwall reviewed (commented):
We already reverted the previous change in 22.1.x and 22.2.x.

These logging changes were driven by context.DeadlineExceeded errors that a customer was having in 22.1.x that we did not know the source of because of no stack trace: https://github.com/cockroachlabs/support/issues/2017

I think this PR should be backported otherwise 22.1.x and 22.2.x won't have stack traces for these errors.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-:-NNlstC27tmNdxYHOr7q:bnig3l2)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @andreimatei and @rafiss)



<!-- Sent from Reviewable.io -->

ecwall commented:
bors r=andreimatei

arulajmani reviewed (commented):
> Up to you, but I feel like the amount of changes has gone overboard for little chance of actually helping anybody.

@andreimatei I see you write this below, and I'd like to talk about this a bit more. This PR skewed with some datadriven tests I'm writing and the way these errors look now is a bit verbose. Consider what I had to update my test to:

```
exec-sql-tenant
CREATE TABLE t(a INT)
----
ok

exec-privileged-op-tenant
ALTER TABLE t SPLIT AT VALUES (0)
----
pq: ba: AdminSplit [/Tenant/10/Table/104/1/0,/Min) RPC error: rpc error: code = Unauthenticated desc = tenant 10 does not have admin split capability
```

Do we absolutely need this change? Could this "original problem" the commit message references be instead solved by verbose logging, and selectively turning it on, or the like? 

I'm worried adding such intense wrapping on so many error paths doesn't really help anyone in the long run. 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96659#-:-NNnVti-CEtmzRsrtLsJ:b-vnneul)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)



<!-- Sent from Reviewable.io -->
